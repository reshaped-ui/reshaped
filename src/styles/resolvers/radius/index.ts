import { responsiveClassNames } from "utilities/props";
import * as T from "styles/types";
import s from "./radius.module.css";

const getRadiusStyles: T.StyleResolver<T.Radius> = (value) => {
	if (!value) return {};

	return {
		classNames: [s.root, ...responsiveClassNames(s, "--radius", value)],
	};
};

export default getRadiusStyles;
