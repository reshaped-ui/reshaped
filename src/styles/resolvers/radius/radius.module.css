.root {
	border-radius: var(--rs-radius);
}

@responsive .--radius {
	@value none {
		--rs-radius: 0;
	}

	@value small {
		--rs-radius: var(--rs-radius-small);
	}

	@value medium {
		--rs-radius: var(--rs-radius-medium);
	}

	@value large {
		--rs-radius: var(--rs-radius-large);
	}

	@value circular {
		--rs-radius: 9999px;
	}
}
