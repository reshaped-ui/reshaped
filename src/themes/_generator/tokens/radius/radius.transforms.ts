import type * as T from "./radius.types";
import type { Transformer } from "../types";

const transformedToken: Transformer<T.Token> = (name, token) => [
	{
		name,
		tokenType: "radius",
		type: "variable",
		value: `${token.px}px`,
	},
];

export default transformedToken;
