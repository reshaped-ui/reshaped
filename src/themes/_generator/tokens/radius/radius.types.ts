export type Name = "small" | "medium" | "large";
export type Token = { px: number };
