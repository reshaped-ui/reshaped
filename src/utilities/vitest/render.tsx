import { render as vitestRender } from "vitest-browser-react";
import Reshaped from "components/Reshaped";
import "themes/reshaped/theme.css";

const Wrapper = (props: { children: React.ReactNode }) => {
	return <Reshaped>{props.children}</Reshaped>;
};

const render = (node: React.ReactNode) => {
	const screen = vitestRender(<div id="root">{node}</div>, {
		wrapper: Wrapper,
	});

	return { ...screen, container: document.getElementById("root")! };
};

export default render;
