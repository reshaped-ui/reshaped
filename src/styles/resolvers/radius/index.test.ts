import { expect, test, describe } from "vitest";
import radius from "./index";

describe("Styles/Radius", () => {
	test("handles value", () => {
		expect(radius("large")).toMatchSnapshot();
	});

	test("handles none", () => {
		expect(radius("none")).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(radius()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(radius({ s: "small", m: "none", l: "circular" })).toMatchSnapshot();
	});
});
