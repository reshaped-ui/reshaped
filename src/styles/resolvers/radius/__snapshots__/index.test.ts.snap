// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Styles/Radius > handles none 1`] = `
{
  "classNames": [
    "_root_9458de",
    "_--radius-none_9458de",
  ],
}
`;

exports[`Styles/Radius > handles responsive value 1`] = `
{
  "classNames": [
    "_root_9458de",
    "_--radius-small_9458de",
    "_--radius-none--m_9458de",
    "_--radius-circular--l_9458de",
  ],
}
`;

exports[`Styles/Radius > handles undefined value 1`] = `null`;

exports[`Styles/Radius > handles value 1`] = `
{
  "classNames": [
    "_root_9458de",
    "_--radius-large_9458de",
  ],
}
`;
